bin/circleci-docker-build.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+set -ex
+
+DOCKER_CACHE_PATH=~/docker
+DOCKER_CACHE_FILE="${DOCKER_CACHE_PATH}/image.tgz"
+
+mkdir -p $DOCKER_CACHE_PATH
+make clean
+
+if [[ -f $DOCKER_CACHE_FILE ]]; then
+  gunzip -c "$DOCKER_CACHE_FILE" | docker load;
+fi
+
+if [[ -f $DOCKER_CACHE_PATH/bedrock.db ]]; then
+  mv $DOCKER_CACHE_PATH/bedrock.db ./
+fi
+
+LOCALES_TAR_FILE="$DOCKER_CACHE_PATH/locales.tgz"
+if [[ -f "$LOCALES_TAR_FILE" ]]; then
+  tar xzf "$LOCALES_TAR_FILE"
+  rm -f "$LOCALES_TAR_FILE"
+fi
+
+MFSA_TAR_FILE="$DOCKER_CACHE_PATH/mfsa_repo.tgz"
+if [[ -f "$MFSA_TAR_FILE" ]]; then
+  tar xzf "$MFSA_TAR_FILE"
+  rm -f "$MFSA_TAR_FILE"
+fi
+
+PD_TAR_FILE="$DOCKER_CACHE_PATH/pd_files.tgz"
+if [[ -f "$PD_TAR_FILE" ]]; then
+  tar xzf "$PD_TAR_FILE"
+  rm -f "$PD_TAR_FILE"
+fi
+
+if [[ "$CIRCLE_BRANCH" == demo__* ]]; then
```

Comment:
The idea behind caching that stuff at all is that it'll be faster to update for demo builds. PR builds don't really need the files, but they can use the cache if it's there. It's possible we could skip the extraction for PR builds and it would work fine too.